homeassistant/components/device_tracker/cppm_tracker.py

```diff
@@ -0,0 +1,85 @@
+"""
+Support for ClearPass Policy Manager.
+
+Allows tracking devices with CPPM.
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    PLATFORM_SCHEMA, DeviceScanner, DOMAIN
+)
+from homeassistant.const import (
+    CONF_HOST, CONF_API_KEY
+)
+
+REQUIREMENTS = ['clearpasspy==1.0.2']
+
+SCAN_INTERVAL = timedelta(seconds=120)
+
+CLIENT_ID = 'client_id'
+
+GRANT_TYPE = 'client_credentials'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CLIENT_ID): cv.string,
+    vol.Required(CONF_API_KEY): cv.string,
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_scanner(hass, config):
+    """Initialize Scanner."""
+    from clearpasspy import ClearPass
+    data = {
+        'server': config[DOMAIN][CONF_HOST],
+        'grant_type': GRANT_TYPE,
+        'secret': config[DOMAIN][CONF_API_KEY],
+        'client': config[DOMAIN][CLIENT_ID]
+    }
+    cppm = ClearPass(data)
+    if cppm.access_token is not None:
+        new_scan = CPPMDeviceScanner(config[DOMAIN], cppm)
+        return new_scan
+    return None
+
+
+class CPPMDeviceScanner(DeviceScanner):
+    """Initialize class."""
+
+    def __init__(self, config, cppm):
+        """Initialize class."""
+        self._cppm = cppm
+        self.results = None
+
+    def scan_devices(self):
+        """Initialize scanner."""
+        self.get_cppm_data()
+        return [device['mac'] for device in self.results]
+
+    def get_device_name(self, device):
+        """Retrieve device name."""
+        return [device['name'] for device in self.results]
```

Comment:
```suggestion
        name = next((
            result['name'] for result in self.results
            if result['mac'] == device), None)
        return name
```

Reply:
For the sake of me learning something...what is the difference in using `next()` vs `for()`

Reply:
https://docs.python.org/3/library/functions.html#next

Since we only want a single item, and not a list, we use `next` + generator expression. Generator expression doesn't create the whole list, so is more efficient.

Reply:
Thanks. I'll do some reading on this.